sdk/search/azure-search/src/main/java/com/azure/search/SearchServiceClientBuilder.java

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.search;
+
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.logging.ClientLogger;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * This class provides a fluent builder API to help aid the configuration and instantiation of
+ * {@link SearchServiceClient SearchServiceClients} and {@link SearchServiceAsyncClient SearchServiceAsyncClients}.
+ * Call {@link #buildClient() buildClient} and {@link #buildAsyncClient() buildAsyncClient} respectively to construct
+ * an instance of the desired client.
+ *
+ * <p>
+ * The following information must be provided on this builder:
+ *     <ul>
+ *         <li>the Azure Cognitive Search service endpoint through {@code .endpoint()}
+ *         <li>the API key through {@code .credential()}</li>
+ *     </ul>
+ * </p>
+ */
+@ServiceClientBuilder(serviceClients = {SearchServiceClient.class, SearchServiceAsyncClient.class})
+public class SearchServiceClientBuilder extends SearchClientBuilder {
+    private final ClientLogger logger = new ClientLogger(SearchServiceClientBuilder.class);
+
+    /**
+     * Default Constructor
+     */
+    public SearchServiceClientBuilder() {
+        init();
+    }
+
+    /**
+     * Sets the api version to work against
+     *
+     * @param apiVersion api version
+     * @return the updated SearchServiceClientBuilder object
+     */
+    public SearchServiceClientBuilder apiVersion(SearchServiceVersion apiVersion) {
+        this.apiVersion = apiVersion;
+        return this;
+    }
+
+    /**
+     * Sets the Azure Cognitive Search service endpoint
+     *
+     * @param endpoint the endpoint URL to the Azure Cognitive Search service
+     * @return the updated SearchIndexClientBuilder object
+     */
+    public SearchServiceClientBuilder endpoint(String endpoint) {
```

Comment:
Generally endpoint is checked for being null and if it is an exception is thrown.